### DISCUSSION TITLE
Adding CPU workers to dataloader

### DISCUSSION THREAD

**Discussion by ADR3-N** [voicepaw/so-vits-svc-fork#534] [General]
I was reading that CPUs are not too shabby for natural language processing applications such as what we are doing with so-vits-svc. I also read that it's impossible to change the number of workers for a dataloader. You have to create a new one and then tell the application to load the new one with a defined quantity of workers.

I have been searching around and I suppose using the wrong engine. It was only after I used Bing AI that it explained to me I would need a new one, but didn't know how to write the code for so-vits to recognize the new dataloader and use it.

How would I go about that with so-vits? Which parameter would I change, that is calling the dataloader, meaning. Windows 10, Python.

**Comment by 34j** [comment]
I have no idea what you are trying to say. num_workers can be changed via config file.

**Reply by ADR3-N** [reply]
Whaaat? You've got to be kidding me. I must be looking at the wrong config file then, as I see no specification in the configs/44k/config.json. Will keep looking. If I have not thumbed you up, I haven't found it yet

**Reply by ADR3-N** [reply]
Error I am getting:

[02:31:50] C:\Users\lolne\anaconda3\envs\so-vits-fork-new\lib\site-packages\lightning\pytorch\trainer\connectors\data_connector.py:430: PossibleUserWarning: The dataloader,   warnings.py:109
                    train_dataloader, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` (try 12 which is the number of cpus on
                    this machine) in the `DataLoader` init to improve performance.

**the only problem is, I have NO idea where this dataloader init or config file would be stored, and there are thousands of init files to trawl through**

Edit: I have found the dataloader init this seems to be referring too after trawling through admittedly the exact listed folder structure, but the init itself is ... blank, 0 bytes, last edited 25 days ago. I'm going to guess you can go without it since I have been training just fine, but that it is ideally not supposed to be blank. What could that possibly mean?

**Reply by GarrettConway** [reply]
@ADR3-N It is stored under train as num_workers? Are you using an old version perhaps?
https://github.com/voicepaw/so-vits-svc-fork/blob/71e589d3a81032326bf9f38dee5ce0aa1370e8a9/src/so_vits_svc_fork/train.py#L55

https://github.com/voicepaw/so-vits-svc-fork/blob/71e589d3a81032326bf9f38dee5ce0aa1370e8a9/src/so_vits_svc_fork/preprocessing/config_templates/so-vits-svc-4.0v1-legacy.json#L23

As a sidenote, in most cases (especially on an ssd) the dataloader is a pretty minor part of the overall time per step. If training is running fine, you can safely ignore that.

**Reply by ADR3-N** [reply]
It's entirely possible that I am in fact running an older version -- also thank you, folder structure helped a lot. If it's true I can just ignore this, that gets a monkey off my back. As it is I've checked both the files you mentioned and there seems to be no corresponding line for either of those, or the file is not being used. I have checked ALL config files and none seems to specify a number of workers beyond the templates. Certainly interesting. Oh well, I can tell a friend as long as the training is humming along, they're fine.

Thanks so much!
